scripts/website/generate.go

```diff
@@ -143,6 +143,68 @@ func main() {
 		}
 	}
 
+	// mutation
+	mutationPath := filepath.Join(rootDir, mutationEntryPoint)
+	mutationDirEntry, err := os.ReadDir(mutationPath)
+	if err != nil {
+		fmt.Println("error while listing directories under mutation")
+		panic(err)
+	}
+
+	for _, entry := range mutationDirEntry {
+		if entry.Type().IsDir() {
+			basePath, _ := filepath.Abs(filepath.Join(mutationPath, entry.Name()))
+			directories, err := os.ReadDir(basePath)
+			if err != nil {
+				fmt.Println("error while listing directories under ", entry.Name())
+				panic(err)
+			}
+
+			for _, dir := range directories {
+				if dir.Type().IsDir() {
+					fmt.Println("Generating markdown for ", filepath.Join(basePath, dir.Name()))
+				}
+
+				// get all files with name starting with "mutation"
+				files, err := os.ReadDir(filepath.Join(basePath, dir.Name(), "samples"))
+				if err != nil {
+					panic(err)
+				}
+
+				mutationTemplateContent, err := os.ReadFile(filepath.Join(pwd, "mutation-template.md"))
+				if err != nil {
+					fmt.Println("error while reading mutation-template.md")
+					panic(err)
+				}
+
+				for _, file := range files {
+					var fileContentBytes []byte
+					if strings.HasPrefix(file.Name(), "mutation") {
+						// read mutation.yaml
+						fileContentBytes, err = os.ReadFile(filepath.Join(basePath, dir.Name(), "samples", file.Name()))
+						if err != nil {
+							fmt.Println("error while reading ", file.Name())
+							panic(err)
+						}
+
+						replacer := strings.NewReplacer(
+							"%RAWURL%", sourceURL+filepath.Join(mutationEntryPoint, entry.Name(), dir.Name(), "samples", file.Name()),
+							"%EXAMPLES%", fmt.Sprintf("%s", fileContentBytes),
+							"%TITLE%", dir.Name(),
+							"%FILENAME%", fmt.Sprintf("mutation-%s", dir.Name()),
+						)
+
+						os.WriteFile(
+							filepath.Join(rootDir, "website/docs", fmt.Sprintf("mutation-%s.md", dir.Name())),
```

Comment:
IMO, I think it's cleaner if we put all the existing validation pages under `website/docs/validation` and new mutation pages under `website/docs/mutation-examples` so the url/navigation is self-documenting. wdyt?

Reply:
+1

Reply:
make sense. updates